src/core/client/admin/components/StoryInfoDrawer/ModerateStoryButton.tsx

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable */
+import { Localized } from "@fluent/react/compat";
+import React, { FunctionComponent, useMemo } from "react";
+import { graphql } from "react-relay";
+
+import { Button } from "coral-ui/components/v3/Button/Button";
+import { getModerationLink } from "coral-framework/helpers";
+import { withFragmentContainer, useLocal } from "coral-framework/lib/relay";
+import { Ability, can } from "coral-framework/permissions";
+
+import { ModerateStoryButton_local } from "coral-admin/__generated__/ModerateStoryButton_local.graphql";
+import { ModerateStoryButton_settings } from "coral-admin/__generated__/ModerateStoryButton_settings.graphql";
+import { ModerateStoryButton_story } from "coral-admin/__generated__/ModerateStoryButton_story.graphql";
+import { ModerateStoryButton_viewer } from "coral-admin/__generated__/ModerateStoryButton_viewer.graphql";
+
+import * as styles from "./ModerateStoryButton.css";
+
+export interface Props {
+  settings: ModerateStoryButton_settings;
+  viewer: ModerateStoryButton_viewer;
+  story: ModerateStoryButton_story;
+}
+
+const ModerateStoryButton: FunctionComponent<Props> = ({
+  settings,
+  story: { id, isArchived, isArchiving, canModerate },
+  viewer
+}) => {
+  const [{ accessToken }] = useLocal<ModerateStoryButton_local>(graphql`
+    fragment ModerateStoryButton_local on Local {
+      accessToken
+    }
+  `);
+
+  const href = useMemo(() => {
+    let link = getModerationLink({ storyID: id });
```

Comment:
There is a bit of duplicated between this and the stream-side ModerateStreamConainer, but putting that compnent in a shared directory was tricky because of the differing `Local` schema, and so I went with duplicating it for now.